lib/RexStanSettings.php

```diff
@@ -0,0 +1,250 @@
+        'Deprecation Warnings' => 'vendor/phpstan/phpstan-deprecation-rules/rules.neon',
+        'PHPUnit' => 'vendor/phpstan/phpstan-phpunit/rules.neon',
+        'phpstan-dba' => 'lib/phpstan-dba.neon',
+        'cognitive complexity' => 'lib/cognitive-complexity.neon',
+    ];
+
+    /**
+     * ... und der Links zu Erläuterungen. ['label' => 'url'].
+     *
+     * @var array<string>
+     */
+    private static $phpstanExtensionDocLinks = [
+        'Bleeding-Edge' => 'https://phpstan.org/blog/what-is-bleeding-edge',
+        'Strict-Mode' => 'https://github.com/phpstan/phpstan-strict-rules#readme',
+        'Deprecation Warnings' => 'https://github.com/phpstan/phpstan-deprecation-rules#readme">Deprecation-Warnings',
+        'PHPUnit' => 'https://github.com/phpstan/phpstan-phpunit#readme">PHPUnit',
+        'phpstan-dba' => 'https://staabm.github.io/archive.html#phpstan-dba',
+        'cognitive complexity' => 'https://tomasvotruba.com/blog/2018/05/21/is-your-code-readable-by-humans-cognitive-complexity-tells-you/',
+    ];
+
+    /**
+     * Liste der derzeit relevanten PHP-Versionen. [versionId => 'label'].
+     *
+     * @var array<string>
+     */
+    private static $phpVersionList = [
+        70300 => '7.3 [Mindestanforderung für REDAXO]',
+        70400 => '7.4',
+        80000 => '8.0',
+        80100 => '8.1',
+        80200 => '8.2',
+    ];
+
+    /**
+     * Initialisiert das Formular.
+     *
+     * @return void
+     */
+    public function init()
+    {
+        $extensions = [];
+        foreach (self::$phpstanExtensions as $label => $path) {
+            $extensions[rex_path::addon('rexstan', $path)] = $label;
+        }
+
+        $extensionLinks = [];
+        foreach (self::$phpstanExtensionDocLinks as $label => $link) {
+            $extensionLinks[] = '<a href="'.$link.'">'.$label.'</a>';
+        }
+
+        /**
+         * Liste der analysierbaren Verzeichnisse (= Addons) ermitteln
+         * Sonderfall: developer-Addon.
+         */
+
+        $scanTargets = [];
+        foreach (rex_addon::getAvailableAddons() as $availableAddon) {
+            $scanTargets[$availableAddon->getPath()] = $availableAddon->getName();
+
+            if ('developer' === $availableAddon->getName() && class_exists(rex_developer_manager::class)) {
+                $scanTargets[rex_developer_manager::getBasePath() .'/modules/'] = 'developer: modules';
+                $scanTargets[rex_developer_manager::getBasePath() .'/templates/'] = 'developer: templates';
+            }
+        }
+
+        /**
+         * Die aktuelle PHP-Version des Servers (SAPI) markieren.
+         */
+
+        $version = (int) (PHP_VERSION_ID / 100) * 100;
+        $phpVersions = self::$phpVersionList;
+        if (isset($phpVersions[$version])) {
+            $phpVersions[$version] .= ' [aktuelle Webserver-Version (SAPI)]';
+        }
+
+        $field = $this->addInputField('number', 'level', null, ['class' => 'form-control', 'min' => 0, 'max' => 9]);
```

Comment:
das config form hier in einer methode zu bauen kann so bleiben, statt `$this` könnte man einfacher `$form = rex_config_form::factory('rexstan');` als lokale variable verwenden

Reply:
Da kann ich grad nicht folgen. $this ist doch bereits das Formular. Was hilft es, innerhalb der Init-Methode des schon angelegten Formulars ein weiteres Formular anzulegen, das zudem nur in der Init-Methode existiert.